bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -485,14 +481,14 @@ void ReplicaImp::onMessage<ClientRequestMsg>(std::unique_ptr<ClientRequestMsg> m
         LOG_WARN(CNSUS,
                  "ClientRequestMsg dropped. Primary request queue is full. "
                      << KVLOG(clientId, reqSeqNum, requestsQueueOfPrimary.size()));
-        delete msg;
+
         return;
       }
       if (clientsManager->canBecomePending(clientId, reqSeqNum)) {
         LOG_DEBUG(CNSUS, "Pushing to primary queue, request " << KVLOG(reqSeqNum, clientId, senderId));
         if (time_to_collect_batch_ == MinTime) time_to_collect_batch_ = getMonotonicTime();
         metric_primary_batching_duration_.addStartTimeStamp(cid);
-        requestsQueueOfPrimary.push(msg);
+        requestsQueueOfPrimary.push(m.release());
```

Comment:
I recommend to check how much work it will be to do a std::move here and convert the container to hold unique_ptrs.
If it's too much work as part of the current task, lets open a new task to do so (lets not keep this "fall between the chairs").
From a 1st look, it does look like more work compare to previous container, but I'm not 100% sure.
Anyhow,, lets discuss that before you finalize the task.